#### A Test to See How Our Best Models Predict Delays with the Real Data from the Second Quarter 2019!

In [2]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
%matplotlib inline

import seaborn as sns
import time

import warnings
warnings.simplefilter('ignore')

# RNG used for seeding
rng = int(np.random.randint(low=1, high=2000, size=1))

In [7]:
# Best models of all techniques tested
Best_model_outcomes = {
    "Machine Learning Procedure": ["Linear Regression", "", "Logistic Regression", "", "", "Decision Tree", "Random Forest", \
                     "", "Stochastic Gradient Boosting", "", "", "", "Stochastic Gradient Descending", ""],
    
    
    "Model Variable": ["Depature Delay", "Arrival Delay", \
              "Departure Delay", "Departure Delay Over 15 Minutes", "Arrival Delay", \
              "Departure Delay", "Arrival Delay", "Arrival Delay 3 Categories", \
              "Departure Delay Over 15 Minutes", "Departure Delay 3 Categories", \
              "Arrival Delay", "Arrival Delay Over 15 Minutes", \
              "Departure Delay Over 15 Minutes", "Arrival Delay Over 15 Minutes"],
    
    
    "Model Type": ["Linear Regression", "Linear Regression", \
                    "Logistic Regression", "Logistic Regression", "Logistic Regression", \
                    "Decision Tree + PCA", "Random Forest", "Random Forest", \
                    "Scaled Stochastic Gradient Boosting", "Scaled Stochastic Gradient Boosting", \
                    "Scaled Stochastic Gradient Boosting", "Scaled Stochastic Gradient Boosting", \
                    "Scaled Stochastic Gradient Descending + PCA", "Scaled Stochastic Gradient Descending"],
    
    
    "Features": [19, 19, 10, 10, 18, 10, 18, 18, 10, 23, 26, 26, 10, 33],
    
    "Test Data R2": [.9542, .9537, .9586, .99, .8854, .98665, .843, .8586, .99, .8121, .8895, .9977, .99, .8373]
    
                }

Best_model_outcomes_df = pd.DataFrame(Best_model_outcomes)
Best_model_outcomes_df

Machine Learning Procedure                   Model Variable  \
0                Linear Regression                   Depature Delay   
1                                                     Arrival Delay   
2              Logistic Regression                  Departure Delay   
3                                   Departure Delay Over 15 Minutes   
4                                                     Arrival Delay   
5                    Decision Tree                  Departure Delay   
6                    Random Forest                    Arrival Delay   
7                                        Arrival Delay 3 Categories   
8     Stochastic Gradient Boosting  Departure Delay Over 15 Minutes   
9                                      Departure Delay 3 Categories   
10                                                    Arrival Delay   
11                                    Arrival Delay Over 15 Minutes   
12  Stochastic Gradient Descending  Departure Delay Over 15 Minutes   
13                                    Arrival Delay Over 15 Minutes   

                                     Model Type  Features  Test Data R2  
0                             Linear Regression        19       0.95420  
1                             Linear Regression        19       0.95370  
2                           Logistic Regression        10       0.95860  
3                           Logistic Regression        10       0.99000  
4                           Logistic Regression        18       0.88540  
5                           Decision Tree + PCA        10       0.98665  
6                                 Random Forest        18       0.84300  
7                                 Random Forest        18       0.85860  
8           Scaled Stochastic Gradient Boosting        10       0.99000  
9           Scaled Stochastic Gradient Boosting        23       0.81210  
10          Scaled Stochastic Gradient Boosting        26       0.88950  
11          Scaled Stochastic Gradient Boosting        26       0.99770  
12  Scaled Stochastic Gradient Descending + PCA        10       0.99000  
13        Scaled Stochastic Gradient Descending        33       0.83730

In [2]:
# Best models for the 6 tested dependent variables.
# 4 over 95%, the other two over 89% and 81%.  Seems like it is time to compare with the real data.
Best_model = {
    "Model Variable": ["Depature Delay", "Departure Delay Over 15 Minutes", "Departure Delay 3 Categories", "", \
              "Arrival Delay", "Arrival Delay", "Arrival Delay Over 15 Minutes", "Arrival Delay 3 Categories", ""],
    
    
    
    "Model Type": ["Linear Regression", "Scaled Stochastic Gradient Descending + PCA", \
                   "Scaled Stochastic Gradient Boosting", "", "Linear Regression", "Logistic Regression", \
                   "Scaled Stochastic Gradient Boosting", "Logistic Regression", ""],
    
    
    
    "Features": [10, 10, 23, "", 16, 18, 26, 32, ""],
    
    "Test Data R2": [.9431, .99, .8121, "", .9537, .8847, .9977, .8981, ""],
    
    "Second Quarter R2": [.94, .9977, .5776, "", .9609, .5172, .7079, .5088, ""]
    
                }

Best_model_df = pd.DataFrame(Best_model)
Best_model_df

Model Variable  \
0                   Depature Delay   
1  Departure Delay Over 15 Minutes   
2     Departure Delay 3 Categories   
3                                    
4                    Arrival Delay   
5                    Arrival Delay   
6    Arrival Delay Over 15 Minutes   
7       Arrival Delay 3 Categories   
8                                    

                                    Model Type Features Test Data R2  \
0                            Linear Regression       10       0.9431   
1  Scaled Stochastic Gradient Descending + PCA       10         0.99   
2          Scaled Stochastic Gradient Boosting       23       0.8121   
3                                                                      
4                            Linear Regression       16       0.9537   
5                          Logistic Regression       18       0.8847   
6          Scaled Stochastic Gradient Boosting       26       0.9977   
7                          Logistic Regression       32       0.8981   
8                                                                      

  Second Quarter R2  
0              0.94  
1            0.9977  
2            0.5776  
3                    
4            0.9609  
5            0.5172  
6            0.7079  
7            0.5088  
8

In [2]:
# Get the second quarter in for compare with the best models
delays_df = pd.read_csv("Delay_second_quarter.csv")

#### All of the following cells are additional cleaning and variables added so the second quarter will match up with the first quarter.  Utilizing it as X_test should not create any issues after these are done.  

In [3]:
# Blank all Nan rows
delays_df = delays_df.fillna(0)

In [4]:
# Dummy variables for east/west coast.  Fixed to use longitude.

delays_df['EAST_COAST_ORIGIN'] = 1*np.ravel(delays_df["ORIGIN_LONGITUDE"] >= -83)
delays_df['EAST_COAST_DEST'] = 1*np.ravel(delays_df["DEST_LONGITUDE"] >= -83)
delays_df['WEST_COAST_ORIGIN'] = 1*np.ravel(delays_df["ORIGIN_LONGITUDE"] <= -114)
delays_df['WEST_COAST_DEST'] = 1*np.ravel(delays_df["DEST_LONGITUDE"] <= -114)

In [5]:
# pd.get_dummies was using too much memory so the following methods are used in its place for dummy variables.
# Create more dummy variables for categorical data:

Weekday = {
           "Monday": 1,
           "Tueday": 2, 
           "Wednesday": 3, 
           "Thursday" : 4,
           "Friday": 5,
           "Saturday": 6,
           "Sunday": 7
          }

Airline = {
        "UA": 1,
        "AA": 2,
        "9E": 3,
        "B6": 4,
        "EV": 5,
        "F9": 6,
        "G4": 7,
        "HA": 8,
        "MQ": 9,
        "NK": 10,
        "OH": 11,
        "OO": 12,
        "VX": 13,
        "WN": 14,
        "YV": 15,
        "YX": 16,
        "AS": 17,
        "DL": 18
}

In [6]:
# Create the weekday dummy variable
delays_df['WEEKDAY_DUMMY'] = delays_df['WEEKDAY'].apply(  \
                            lambda x: next((y for z, y in Weekday.items() if x in z), 0))

In [7]:
# Create the airline dummy variable
delays_df['AIRLINE_DUMMY'] = delays_df['OP_CARRIER'].apply(  \
                            lambda x: next((y for z, y in Airline.items() if x in z), 0))

In [8]:
# Create a categorical departure delay dummy variable.
delay_dict = {0: "On Time", 1: "Delayed"}
delays_df["DEPARTURE_DELAY_DUMMY"]=delays_df["DEPARTURE_DELAY"].copy()
delays_df["DEPARTURE_DELAY"].replace(delay_dict, inplace = True)

In [9]:
# Create a categorical arrival delay dummy variable.
arrive_dict = {0: "On Time", 1: "Delayed"}
delays_df["ARRIVAL_DELAY_DUMMY"]=delays_df["ARRIVAL_DELAY"].copy()
delays_df["ARRIVAL_DELAY"].replace(delay_dict, inplace = True)

In [10]:
# Dictionary for on time, small delay of less than 15 minutes and long delay of over 15 minutes.
Delay_dict = {range(-4000, 1): "On Time",
               range(1, 16): 'Small Delay', 
               range(16, 4000): 'Long Delay',}

In [11]:
# Apply the delay dictionary to categorize departing and arriving delays.
# These both take quite a bit of time to process but use very little system resources.  
# Such is the tradeoff of dealing with big data...

start = time.time()
delays_df['DEPARTURE_DELAY_TEST'] = delays_df['DEP_DELAY'].apply(  \
                            lambda x: next((y for z, y in Delay_dict.items() if x in z), 0))
end = time.time()
print(f"{end-start} seconds")

538.5294439792633 seconds


In [12]:
start = time.time()
delays_df['ARRIVAL_DELAY_TEST'] = delays_df['ARR_DELAY'].apply(  \
                            lambda x: next((y for z, y in Delay_dict.items() if x in z), 0))
end = time.time()
print(f"{end-start} seconds")

544.191000699997 seconds


In [3]:
# Output as new csv.
delays_df.to_csv("Delay_second_quarter2.csv",index=False)

NameError: name 'delays_df' is not defined

In [14]:
# Check to make sure everything worked
delays_df2 = pd.read_csv("Delay_second_quarter2.csv")

In [15]:
delays_df2.head()

FL_DATE  DAY  MONTH    WEEKDAY OP_CARRIER            AIRLINE  \
0  2018-04-01    1      4     Sunday         F9  Frontier Airlines   
1  2018-04-02    2      4     Monday         F9  Frontier Airlines   
2  2018-04-02    2      4     Monday         F9  Frontier Airlines   
3  2018-04-03    3      4    Tuesday         F9  Frontier Airlines   
4  2018-04-04    4      4  Wednesday         F9  Frontier Airlines   

   OP_CARRIER_FL_NUM ORIGIN                                ORIGIN_AIRPORT  \
0                459    MSP  Minneapolis-Saint Paul International Airport   
1                463    MSP  Minneapolis-Saint Paul International Airport   
2                459    MSP  Minneapolis-Saint Paul International Airport   
3                459    MSP  Minneapolis-Saint Paul International Airport   
4                459    MSP  Minneapolis-Saint Paul International Airport   

   ORIGIN_CITY  ... WEATHER_DELAY  NAS_DELAY  SECURITY_DELAY  \
0  Minneapolis  ...           0.0        0.0             0.0   
1  Minneapolis  ...           5.0       77.0             0.0   
2  Minneapolis  ...           0.0        0.0             0.0   
3  Minneapolis  ...           0.0       42.0             0.0   
4  Minneapolis  ...           0.0        0.0             0.0   

   LATE_AIRCRAFT_DELAY  WEEKDAY_DUMMY  AIRLINE_DUMMY  DEPARTURE_DELAY_DUMMY  \
0                  0.0              7              6                      0   
1                  0.0              1              6                      1   
2                  0.0              1              6                      0   
3                  0.0              0              6                      1   
4                  0.0              3              6                      0   

  ARRIVAL_DELAY_DUMMY  DEPARTURE_DELAY_TEST  ARRIVAL_DELAY_TEST  
0                   0               On Time             On Time  
1                   1           Small Delay          Long Delay  
2                   1               On Time         Small Delay  
3                   1           Small Delay          Long Delay  
4                   0               On Time             On Time  

[5 rows x 65 columns]

In [17]:
len(delays_df2)

1816125

In [4]:
# Get the first and second quarter in for comparision.
delays_df = pd.read_csv("Delay_first_quarter1.csv")

In [11]:
delays_df2 = pd.read_csv("Delay_second_quarter2.csv") 

#### BEST DEPARTURE DELAY MODEL -> DEPARTURE DELAY TIMES

#### LINEAR REGRESSION WITH 17 FEATURES (94%)

In [4]:
X = delays_df[["DAY", "MONTH", "CRS_DEP_TIME", "DEPARTURE_TIME_OF_DAY_DUMMY", \
              "OP_CARRIER_FL_NUM", "TAXI_OUT", "WHEELS_OFF", "AIR_TIME", "WHEELS_ON", "TAXI_IN", \
              "CRS_ARR_TIME", "ARRIVAL_TIME_OF_DAY_DUMMY", "CARRIER_DELAY", \
       "WEATHER_DELAY", "NAS_DELAY", "SECURITY_DELAY", "LATE_AIRCRAFT_DELAY"]]

y = delays_df["DEP_DELAY"].values.reshape(-1, 1)
print(X.shape, y.shape)

(1683475, 17) (1683475, 1)


In [5]:
# Scale the data
from sklearn.model_selection import train_test_split
start = time.time()

X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=255)
end = time.time()
print(f"{end-start} seconds")

0.7760090827941895 seconds


In [6]:
from sklearn.preprocessing import StandardScaler

X_scaler = StandardScaler().fit(X_train)
y_scaler = StandardScaler().fit(y_train)

X_train_scaled = X_scaler.transform(X_train)
X_test_scaled = X_scaler.transform(X_test)
y_train_scaled = y_scaler.transform(y_train)
y_test_scaled = y_scaler.transform(y_test)

In [7]:
from sklearn.linear_model import LinearRegression
start = time.time()

model = LinearRegression()
model.fit(X_train_scaled, y_train_scaled)
end = time.time()
print(f"{end-start} seconds")

0.9569876194000244 seconds


In [9]:
# 94.31% is pretty good.
from sklearn.metrics import mean_squared_error
start = time.time()

predictions = model.predict(X_test_scaled)
MSE = mean_squared_error(y_test_scaled, predictions)
r2 = model.score(X_test_scaled, y_test_scaled)
print(f"MSE: {MSE}, R2: {r2}")
end = time.time()
print(f"{end-start} seconds")

MSE: 0.05712342290001597, R2: 0.9431201567906724
0.12295889854431152 seconds


In [10]:
# Use second quarter data
X = delays_df2[["DAY", "MONTH", "CRS_DEP_TIME", "DEPARTURE_TIME_OF_DAY_DUMMY", \
              "OP_CARRIER_FL_NUM", "TAXI_OUT", "WHEELS_OFF", "AIR_TIME", "WHEELS_ON", "TAXI_IN", \
              "CRS_ARR_TIME", "ARRIVAL_TIME_OF_DAY_DUMMY", "CARRIER_DELAY", \
       "WEATHER_DELAY", "NAS_DELAY", "SECURITY_DELAY", "LATE_AIRCRAFT_DELAY"]]

y = delays_df2["DEP_DELAY"].values.reshape(-1, 1)
print(X.shape, y.shape)

(1816125, 17) (1816125, 1)


In [11]:
# Scale the data
from sklearn.model_selection import train_test_split
start = time.time()

X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=rng)
end = time.time()
print(f"{end-start} seconds")

0.8950047492980957 seconds


In [12]:
from sklearn.preprocessing import StandardScaler

X_scaler = StandardScaler().fit(X_train)
y_scaler = StandardScaler().fit(y_train)

X_train_scaled = X_scaler.transform(X_train)
X_test_scaled = X_scaler.transform(X_test)
y_train_scaled = y_scaler.transform(y_train)
y_test_scaled = y_scaler.transform(y_test)

In [13]:
# X_test_scaled and y_test_scaled are from the second quarter while the predictions are from the first quarter.
# 94% accuracy while using the first 100,000 flights from the second quarter.  Not a bad start.

from sklearn.metrics import mean_squared_error
start = time.time()

MSE = mean_squared_error(y_test_scaled[:100000], predictions[:100000])
r2 = model.score(X_test_scaled[:100000], y_test_scaled[:100000])
print(f"MSE: {MSE}, R2: {r2}")
end = time.time()
print(f"{end-start} seconds")

MSE: 1.9068766068323617, R2: 0.9454795172325134
0.056999921798706055 seconds


#### BEST DEPARTURE DELAY OVER 15 MINUTES

#### 10 FEATURES SGBCLASSIFIER PCA SCALED (97.2%)

In [62]:
X = delays_df[["DAY", "MONTH", "DEP_TIME", "DEPARTURE_TIME_OF_DAY_DUMMY", "WEEKDAY_DUMMY", "AIRLINE_DUMMY", \
               "OP_CARRIER_FL_NUM", "TAXI_OUT", "WHEELS_OFF","WEATHER_DELAY",]]

y = delays_df["DEPARTURE_DELAY_OVER_15_MINUTES"].values.reshape(-1, 1)
print(X.shape, y.shape)

(1683475, 10) (1683475, 1)


In [63]:
from sklearn.model_selection import train_test_split

X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=rng)

In [64]:
# Try out PCA if needed
from sklearn.decomposition import PCA
start = time.time()

pca = PCA(.95)
pca.fit(X)
X_train = pca.transform(X_train)
X_test = pca.transform(X_test)
end = time.time()
print(f"{end-start} seconds")

2.3719191551208496 seconds


In [65]:
# Try out a min/max scaler for the data if needed
from sklearn.preprocessing import MinMaxScaler
start = time.time()

scaler = MinMaxScaler()
X_train = scaler.fit_transform(X_train)
X_test = scaler.transform(X_test)
end = time.time()
print(f"{end-start} seconds")

0.34800076484680176 seconds


In [66]:
# SGDClassifier
from sklearn.linear_model import SGDClassifier
from sklearn.model_selection import GridSearchCV
start = time.time()

sgd = SGDClassifier(random_state=rng)
sgd.fit(X_train, y_train)
end = time.time()
print(f"{end-start} seconds")

1.7320270538330078 seconds


In [67]:
# This model seems to hold up.
from sklearn.metrics import accuracy_score
start = time.time()

predictions = sgd.predict(X_test)
y_transposed = (np.transpose(y_test)).flatten()
print(f"Accuracy Score: {accuracy_score(y_transposed, predictions)*100}")
end = time.time()
print(f"{end-start} seconds")

Accuracy Score: 99.00990569512128
0.1609971523284912 seconds


In [68]:
# Set up with second quarter data
X = delays_df2[["DAY", "MONTH", "DEP_TIME", "DEPARTURE_TIME_OF_DAY_DUMMY", "WEEKDAY_DUMMY", "AIRLINE_DUMMY", \
               "OP_CARRIER_FL_NUM", "TAXI_OUT", "WHEELS_OFF","WEATHER_DELAY",]]

y = delays_df2["DEPARTURE_DELAY_OVER_15_MINUTES"].values.reshape(-1, 1)
print(X.shape, y.shape)

(1816125, 10) (1816125, 1)


In [69]:
from sklearn.model_selection import train_test_split

X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=rng)

In [70]:
# Try out PCA if needed
from sklearn.decomposition import PCA
start = time.time()

pca = PCA(.95)
pca.fit(X)
X_train = pca.transform(X_train)
X_test = pca.transform(X_test)
end = time.time()
print(f"{end-start} seconds")

2.4749326705932617 seconds


In [71]:
# Try out a min/max scaler for the data if needed
from sklearn.preprocessing import MinMaxScaler
start = time.time()

scaler = MinMaxScaler()
X_train = scaler.fit_transform(X_train)
X_test = scaler.transform(X_test)
end = time.time()
print(f"{end-start} seconds")

0.371992826461792 seconds


In [73]:
# y_test is using second quarter data while predictions are from the first quarter model.
# This seems a little fishy.  Perhaps look into it more later if possible.  97.274% seems a bit too good.
from sklearn.metrics import accuracy_score
start = time.time()

#predictions = sgd.predict(X_test)
y_transposed = (np.transpose(y_test)).flatten()
print(f"Accuracy Score: {accuracy_score(y_transposed[:100000], predictions[:100000])*100}")
end = time.time()
print(f"{end-start} seconds")

Accuracy Score: 99.274
0.023998022079467773 seconds


In [75]:
pd.crosstab(y_transposed[:100000], predictions[:100000], rownames=["Actual Delays"], colnames=["Predicted Delays"])

Predicted Delays      0
Actual Delays          
0                 99274
1                   726

In [5]:
delays_df_ua = delays_df.loc[delays_df["OP_CARRIER"]=="UA",:]

In [9]:
delays_df_ua2 = delays_df2.loc[delays_df2["OP_CARRIER"]=="UA",:]

NameError: name 'delays_df2' is not defined

#### BEST DEPARTURE DELAY 3 CATEGORIES

#### 23 FEATURES STOCHASTIC GRADIENT BOOSTING SCALED (57.76%)

In [76]:
X = delays_df[["DAY", "MONTH", "CRS_DEP_TIME", "DEPARTURE_TIME_OF_DAY_DUMMY", "AIRLINE_DUMMY",\
              "OP_CARRIER_FL_NUM", "TAXI_OUT", "WHEELS_OFF", "AIR_TIME", "DISTANCE", "WEEKDAY_DUMMY",\
              "ARR_TIME", "CRS_ARR_TIME", "ARRIVAL_TIME_OF_DAY_DUMMY", "CARRIER_DELAY", \
       "WEATHER_DELAY", "NAS_DELAY", "SECURITY_DELAY", "LATE_AIRCRAFT_DELAY",
       "EAST_COAST_ORIGIN", "WEST_COAST_ORIGIN", "EAST_COAST_DEST", "WEST_COAST_DEST"]]

y = delays_df["DEPARTURE_DELAY_TEST"].values.reshape(-1, 1)
print(X.shape, y.shape)

(1683475, 23) (1683475, 1)


In [77]:
# Split for train and test datasets
from sklearn.model_selection import train_test_split
start = time.time()

X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=rng)
end = time.time()
print(f"{end-start} seconds")

1.1299278736114502 seconds


In [78]:
# Try out a min/max scaler for the data
from sklearn.preprocessing import MinMaxScaler

scaler = MinMaxScaler()
X_train = scaler.fit_transform(X_train)
X_test = scaler.transform(X_test)

In [79]:
# Set up and fit the Gradient Booster

from sklearn.ensemble import GradientBoostingClassifier
start = time.time()

gbc = GradientBoostingClassifier(n_estimators=100, random_state=rng)
gbc.fit(X_train, y_train)
end = time.time()
print(f"{end-start} seconds")

1528.1254584789276 seconds


In [80]:
#
from sklearn.metrics import accuracy_score
predictions = gbc.predict(X_test)
prediction_p = gbc.predict_proba(X_test)
y_transposed = (np.transpose(y_test)).flatten()
print(f"Accuracy Score: {accuracy_score(y_transposed, predictions)*100}")

Accuracy Score: 81.37781589995937


In [81]:
# Set with the second quarter data
X = delays_df2[["DAY", "MONTH", "CRS_DEP_TIME", "DEPARTURE_TIME_OF_DAY_DUMMY", "AIRLINE_DUMMY",\
              "OP_CARRIER_FL_NUM", "TAXI_OUT", "WHEELS_OFF", "AIR_TIME", "DISTANCE", "WEEKDAY_DUMMY",\
              "ARR_TIME", "CRS_ARR_TIME", "ARRIVAL_TIME_OF_DAY_DUMMY", "CARRIER_DELAY", \
       "WEATHER_DELAY", "NAS_DELAY", "SECURITY_DELAY", "LATE_AIRCRAFT_DELAY",
       "EAST_COAST_ORIGIN", "WEST_COAST_ORIGIN", "EAST_COAST_DEST", "WEST_COAST_DEST"]]

y = delays_df2["DEPARTURE_DELAY_TEST"].values.reshape(-1, 1)
print(X.shape, y.shape)

(1816125, 23) (1816125, 1)


In [82]:
# Split for train and test datasets
from sklearn.model_selection import train_test_split
start = time.time()

X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=rng)
end = time.time()
print(f"{end-start} seconds")

1.262979507446289 seconds


In [83]:
# Try out a min/max scaler for the data
from sklearn.preprocessing import MinMaxScaler

scaler = MinMaxScaler()
X_train = scaler.fit_transform(X_train)
X_test = scaler.transform(X_test)

In [89]:
# Using predictions for the first quarter model and y_test from the second quarter dataset.
# So this one does not hold up for predictions.

from sklearn.metrics import accuracy_score
# predictions = gbc.predict(X_test)
# prediction_p = gbc.predict_proba(X_test)
y_transposed = (np.transpose(y_test)).flatten()
print(f"Accuracy Score: {accuracy_score(y_transposed[:100000], predictions[:100000])*100}")

Accuracy Score: 57.768


In [86]:
pd.crosstab(y_transposed[:100000], predictions[:100000], rownames=["Actual Delays"], colnames=["Predicted Delays"])

Predicted Delays  Long Delay  On Time  Small Delay
Actual Delays                                     
Long Delay              2348    15576          416
On Time                 8335    55083         1471
Small Delay             2228    14206          337